[Shanmugam Udhaya] iP

UdhayaShan1 opened the pull request:
# RyanGosling ChatBot

> "I drive" - Drive (2011)

RyanGosling bot allows you to keep track of your tasks easily instead of being an unorganised mess! 😊

You can now
- Add todos
- Add deadlines
- Add events
- More (coming soon!)

So how do I become Ryan Gosling?

1. Download the latest release from [here!](https://github.com/UdhayaShan1/ip/releases)
2. Drag the JAR to any directory, recommended a dedicated folder.
3. Run the JAR using java -jar RyanGosling.jar from CMD after changing directory into that folder.

For example, open CMD.
`java -jar RyanGosling.jar`

Simple usage,
```ruby
todo Drive
list
find Drive 
deadline DriveAgain /by 20250523 0900
```

So what **else?**
We have plenty of new features coming soon!

- [x] Functions mentioned before
- [ ] Sequential parsing of user input
- [ ] Modifying task data

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

UdhayaShan1 committed:
Level-0, Renamed bot and added basic printing for greeting using MessagePrinter helper class.

UdhayaShan1 committed:
Level-1, Now will echo commands input by user. Some commands will return personalised replies.

UdhayaShan1 committed:
Level-2, added Storage class to track what commands added.

UdhayaShan1 committed:
Level-3, Refactored the listening to a ChatListener class which abstracts the parsing of the text command as well. Also added Task class and ability to mark/unmark tasks.

UdhayaShan1 committed:
Level-4, created new sub-classes for events.

UdhayaShan1 committed:
A-TextUiTesting, tried text UI testing but not sure if buggy. Will ask and fix if neccesary.

UdhayaShan1 committed:
Level-5, created a new Exception class to handle improper commands. For now, just incomplete todo, deadline, event and unknown command.

UdhayaShan1 committed:
Level-6, added delete functionality.

UdhayaShan1 committed:
A-Enums, created a CommandsEnum enum to store all the possible commands. Changed the checking logic to compare against the enum values.

UdhayaShan1 committed:
Minor edits

UdhayaShan1 committed:
Add save to local text file capability whenever task list is updated

UdhayaShan1 committed:
Add loading capability, throws error if either not in specific format or if file does not exist

UdhayaShan1 committed:
Merge branch 'branch-Level-7'

UdhayaShan1 committed:
Store LocalDates instead of String dates. Removed time handling altogether, requires a big change at the end.

UdhayaShan1 committed:
Irrelevant commit

UdhayaShan1 committed:
Merge branch 'branch-Level-8'

UdhayaShan1 committed:
Refactor code (well not really just renamed already refactored code to fit the new standard)

UdhayaShan1 committed:
Merge branch 'refactor-minimal'

UdhayaShan1 committed:
Organize the codebase into packages

UdhayaShan1 committed:
Merge branch 'refactor-packages'

UdhayaShan1 committed:
Merge branch 'add-gradle-support'

UdhayaShan1 committed:
Update main class of Gradle to run and build

UdhayaShan1 committed:
Date parse test success, added check for correctness, invalid format and logical errors (no feb 31)

UdhayaShan1 committed:
Add unit testing for command parsing, specifically bye and todo correctness

UdhayaShan1 committed:
Merge branch 'j-unit-testing'

UdhayaShan1 committed:
Create jar build

UdhayaShan1 committed:
Merge branch 'jar-build'

UdhayaShan1 committed:
Add Javadoc to most classes and methods, except subclasses of tasks as mostly overridden methods

UdhayaShan1 committed:
Change code standards, especially wrapping

UdhayaShan1 committed:
Merge branch 'branch-A-JavaDoc'

UdhayaShan1 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/datesandtimes/DateTimeParser.java

UdhayaShan1 committed:
Add find capability

Add format check for find and mark/unmark

UdhayaShan1 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/task_lists.txt
#	src/main/java/commands/CommandsParser.java
#	src/main/java/utilities/MessagePrinter.java
#	src/main/java/utilities/TaskList.java

UdhayaShan1 committed:
Minor edit for test and JavaFX setup

UdhayaShan1 committed:
task_list edit, ignore

UdhayaShan1 committed:
Change nothing

UdhayaShan1 committed:
Check if file-path exists, if not help create.

UdhayaShan1 committed:
Merge branch 'change-to-mkdir'

UdhayaShan1 committed:
Add time handling, now appropriate commands must take in time

UdhayaShan1 committed:
Merge branch 'time-implementation'

UdhayaShan1 committed:
Add JavaDoc to some more methods and classes

UdhayaShan1 commented on build.gradle:
```diff
@@ -0,0 +1,61 @@
+    id 'java'
+    id 'application'
+    id 'com.github.johnrengelman.shadow' version '7.1.2'
+}
+
+repositories {
+    mavenCentral()
```
```suggestion
lol
```

UdhayaShan1 reviewed (commented):
Minor changes

UdhayaShan1 committed:
Add CheckStyle config and Gradle check

UdhayaShan1 committed:
Change MessagePrinter and TaskList to return Strings

UdhayaShan1 committed:
Add basic GUI functionality

UdhayaShan1 committed:
Change images to fit bot theme

UdhayaShan1 committed:
Add text color to user input and responses

UdhayaShan1 committed:
Add stage title

UdhayaShan1 committed:
Add introduction message by RyanGosling

UdhayaShan1 committed:
Add help function to get list of commands

UdhayaShan1 committed:
Update JavaDocs to fit the new purpose of class

UdhayaShan1 committed:
Fix bug where list is updated before tasklist is updated

UdhayaShan1 committed:
Merge branch 'A-CheckStyle'

UdhayaShan1 committed:
Merge branch 'Level-10'

# Conflicts:
#	src/main/java/RyanGosling.java
#	src/main/java/commands/CommandsParser.java
#	src/main/java/utilities/MessagePrinter.java
#	src/main/java/utilities/Ui.java

UdhayaShan1 committed:
Minor updates

UdhayaShan1 committed:
Merge branch 'branch-Level-10'

UdhayaShan1 committed:
Rename image variables

rahhulleee commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,125 @@
+     * @return The status icon ("X" if the task is done, " " (space) otherwise).
+     */
+    public String getStatusIcon() {
+        return (this.taskDone ? "X" : " ");
```
i like how you used a ternary operator instead of an If Else block statement to make it seem neater

rahhulleee replied:
I like how you used a ternary operator instead of an if else block to make your code seem neater.

rahhulleee commented on src/main/java/utilities/ResponseHandler.java:
```diff
@@ -0,0 +1,187 @@
+     * @return A personalized reply.
+     */
+    public static String personalisedReply(String message) {
+        HashMap<String, String> responseMap = new HashMap<>();
```
I like your usage of a hashmap, I didn't think of it. It definitely makes the run time faster should there be a lot of responses that need to be stored.

rahhulleee commented on src/main/java/utilities/ResponseHandler.java:
```diff
@@ -0,0 +1,187 @@
+     */
+    public static String markActionPrint(String action, Task task) {
+        StringBuilder stringResponse = new StringBuilder();
+        if (action.equals("mark")) {
```
perhaps you could add a toLowerCase() method to account for cases where user inputs Mark instead of mark

rahhulleee commented on src/main/resources/view/DialogBox.fxml:
```diff
@@ -0,0 +1,16 @@
+<?import javafx.scene.layout.HBox?>
+
+<fx:root alignment="TOP_RIGHT" maxHeight="1.7976931348623157E308" maxWidth="1.7976931348623157E308" prefWidth="400.0" type="javafx.scene.layout.HBox" xmlns="http://javafx.com/javafx/17" xmlns:fx="http://javafx.com/fxml/1">
+    <children>
```
i like you choice of picture. and your readme content as well. very nice !

rahhulleee commented on src/test/java/datesandtimes/DateTimeParserTest.java:
```diff
@@ -0,0 +1,53 @@
+
+        // Check invalid day
+        assertThrows(DateTimeException.class,
+                () -> DateTimeParser.parseDate(LocalDate.parse("2045-05-32")));
```
good test coverage.

rahhulleee reviewed (commented):
Your code is neat and abstracted out into many different classes so overall, good job.

UdhayaShan1 committed:
Fix dialog box to fit the returned message

UdhayaShan1 committed:
Merge branch 'refactor-fxml'

UdhayaShan1 committed:
Add assertions to most methods

All methods all do not have assertion checks.

We should check for basic assumptions of some of the input, trivial
or not, so that runtime can verify them.

We add assertions to check whether input is not null for some methods,
we also check if some of the input equals a specific set of possible
input values. For example, action variable can either be
'mark' or 'unmark'.

Assertions are used as we can terminate the program if we fail to meet
basic assumptions during run-time. These are different from expected
exceptions.

UdhayaShan1 committed:
Add simple switch case to CommandsParser class

UdhayaShan1 committed:
Abstract pattern regex to PatternParser class to adhere to SLAP

CommandsParser class originally included regex check to parse
user input.

It leads to less readable code and violates SLAP due to differing
levels of abstraction. Here, (originally) taskList's methods were
abstracted whereas pattern regex implementation was in this class.

Create new PatternParser class and call from CommandsParser instead.

We do this to adhere to basic OOP by abstracting out the methods.

Another way is to change to non-static and create a instance of
PatternParser instead.

UdhayaShan1 committed:
Clean up dead code and condense some functions

UdhayaShan1 committed:
Add JavaDocs for PatternParser class

UdhayaShan1 committed:
Abstract the pattern regex from Storage class to adhere to SLAP

Currently, storage class has no abstraction for the pattern regex
and implementation to check for regex before adding to list of tasks
is present alongside other abstraction.

This violates SLAP and the method is simply too long.

We abstract the pattern regex to PatternParser as earlier commits did.

We adhere to OOP and make use of an already useful class that is made
for such pattern regex abstraction.

UdhayaShan1 committed:
Merge pull request #1 from UdhayaShan1/branch-A-CodeQuality

Enhance code quality, refer to individual commit for clearer details.

UdhayaShan1 committed:
Merge branch 'master' of https://github.com/UdhayaShan1/ip

UdhayaShan1 committed:
Merge branch 'master' into branch-A-Assertions

UdhayaShan1 committed:
Merge pull request #2 from UdhayaShan1/branch-A-Assertions

Add assertions to most methods

UdhayaShan1 committed:
Merge branch 'master' of https://github.com/UdhayaShan1/ip

UdhayaShan1 committed:
Fix mismatch in variable names

UdhayaShan1 committed:
Add copied GitHub workflow

UdhayaShan1 committed:
Merge branch 'ci-setup'

UdhayaShan1 committed:
Add update feature

Currently, there is no update feature. We are forced to delete and
reinput the task if needed.

This is clearly inconvenient as reinputting is not easy.

Now we can select a index and update its appropriate field without
changing other fields. We do so by simple update command.

Since our constraint is CLI and single line is preferred over
multi lines, we utilize a simple single line update command.

Tests still have to be added however.

UdhayaShan1 committed:
Merge branch 'branch-C-Update'

UdhayaShan1 committed:
Fix all check styles flagged, includes main and test

UdhayaShan1 committed:
Merge pull request #3 from UdhayaShan1/branch-fix_checkstyle

Fix all check styles flagged, includes main and test

UdhayaShan1 committed:
Add differing shape for user and bot display picture

UdhayaShan1 committed:
Add categorizer class and change color of response if response is error

UdhayaShan1 committed:
Simple change

UdhayaShan1 committed:
Merge branch 'branch-A-BetterGui'

UdhayaShan1 committed:
Add new exceptions and SLAP for mark/unmark

UdhayaShan1 committed:
Add SLAP for mark/unmark

UdhayaShan1 committed:
Add spaces check for user input

UdhayaShan1 committed:
Add date time validation check for event tasks

UdhayaShan1 committed:
Merge branch 'branch-A-MoreErrorHandling'

UdhayaShan1 committed:
Fix find pattern bug

Find pattern was wrongly set to event pattern.

UdhayaShan1 committed:
Add error checks to delete command

UdhayaShan1 committed:
Add basic user guide

UdhayaShan1 committed:
Update JavaDocs

UdhayaShan1 committed:
Fix delete command bug, deleted redundant code from stub

UdhayaShan1 committed:
Modify delete command arguments exception message

UdhayaShan1 committed:
Merge branch 'final-TouchUp'

UdhayaShan1 committed:
Add bye command to exit the programme

UdhayaShan1 committed:
Minor edit

UdhayaShan1 committed:
Fix minor checkstyle

UdhayaShan1 committed:
Minor greeting edit
